tests/test_json_encoder.py

```diff
@@ -0,0 +1,15 @@
+from flask.json import JSONEncoder
+
+
+def test_encoder_html():
+    class ObjectWithHTML:
+        TEST_HTML = "<p>test</p>"
+
+        def __html__(self):
+            return self.TEST_HTML
+
+    encoder = JSONEncoder()
```

Comment:
Use `json.dumps` rather than using the encoder directly, as that's not the typical use case.